pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/MeasuresFunctors.hpp

```diff
@@ -16,27 +16,426 @@
+    std::size_t rev_wire_5;
+    std::size_t rev_wire_6;
+    std::size_t rev_wire_7;
+    std::size_t rev_wire_8;
+    std::size_t parity_0;
+    std::size_t parity_1;
+    std::size_t parity_2;
+    std::size_t parity_3;
+    std::size_t parity_4;
+    std::size_t parity_5;
+    std::size_t parity_6;
+    std::size_t parity_7;
+    std::size_t parity_8;
+    std::size_t parity_9;
+
+    getProbsNQubitOpFunctor(
+        const Kokkos::View<ComplexT *> &arr_, const std::size_t num_qubits_,
+        [[maybe_unused]] const std::vector<std::size_t> &wires_)
+        : value_count{1U << wires_.size()}, arr{arr_}, n_wires{wires_.size()} {
+        PL_ABORT_IF(num_wires != 0 && num_wires != n_wires,
+                    "num_wires must be equal to n_wires.");
+        const std::size_t n_wires = wires_.size();
+        std::vector<std::size_t> rev_wires_(n_wires);
+        for (std::size_t k = 0; k < n_wires; k++) {
+            rev_wires_[n_wires - 1 - k] = (num_qubits_ - 1) - wires_[k];
+        }
+        std::vector<std::size_t> parity_ =
+            Pennylane::Util::revWireParity(rev_wires_);
+        if constexpr (num_wires == 0) {
+            rev_wires =
+                Pennylane::LightningKokkos::Util::vector2view(rev_wires_);
+            parity = Pennylane::LightningKokkos::Util::vector2view(parity_);
+        }
+        if constexpr (num_wires > 0) {
+            rev_wire_0 = rev_wires_[0];
+            parity_0 = parity_[0];
+            parity_1 = parity_[1];
+        }
+        if constexpr (num_wires > 1) {
+            rev_wire_1 = rev_wires_[1];
+            parity_2 = parity_[2];
+        }
+        if constexpr (num_wires > 2) {
+            rev_wire_2 = rev_wires_[2];
+            parity_3 = parity_[3];
+        }
+        if constexpr (num_wires > 3) {
+            rev_wire_3 = rev_wires_[3];
+            parity_4 = parity_[4];
+        }
+        if constexpr (num_wires > 4) {
+            rev_wire_4 = rev_wires_[4];
+            parity_5 = parity_[5];
+        }
+        if constexpr (num_wires > 5) {
+            rev_wire_5 = rev_wires_[5];
+            parity_6 = parity_[6];
+        }
+        if constexpr (num_wires > 6) {
+            rev_wire_6 = rev_wires_[6];
+            parity_7 = parity_[7];
+        }
+        if constexpr (num_wires > 7) {
+            rev_wire_7 = rev_wires_[7];
+            parity_8 = parity_[8];
+        }
+        if constexpr (num_wires > 8) {
+            rev_wire_8 = rev_wires_[8];
+            parity_9 = parity_[9];
+        }
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void init(PrecisionT dst[]) const {
+        for (unsigned i = 0; i < value_count; ++i)
+            dst[i] = 0.0;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void join(PrecisionT dst[], const PrecisionT src[]) const {
+        for (unsigned i = 0; i < value_count; ++i)
+            dst[i] += src[i];
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void apply0(const std::size_t i0, const std::size_t rev_wire,
```

Comment:
Compute the indices `ixx...xx0` and `ixx...xx1` and accumulate into `dst` (`probs`).